### PR TITLE
[WFCORE-3995] forbid invalid Sensitivity Constraint attribute read-wr…

### PR DESCRIPTION
**Pull request by soul2zimate**
…ite combinations.

issues: https://issues.jboss.org/browse/WFCORE-3995 and https://issues.jboss.org/browse/WFCORE-4061

### CODE REVIEW

**Comment by bstansberry** [comment]
There's a couple problems in this area:

1) SensitivityClassificationWriteAttributeHandler does not have any rollback handling.  That's not a problem in this PR; it's just this PR made me realize it.  I suspect other handlers like this may have the same issue.  I filed https://issues.jboss.org/browse/WFCORE-4061

2) Cross-attribute validation like what's done here must be done in an added step, as that allows composite ops to work without undue concern about order of steps in the composite. Without that this might break existing scripts.  (It would be ok to change the AbstractSensitivity and then add the validation step, counting on the rollback to correct the AbstractSensitivity if the validation fails. But WFCORE-4061 would have to be corrected for that to be a correct approach.)

**Comment by soul2zimate** [comment]
I have rebased to

- add rollback handling first for WFCORE-4061
- remove some redundant code because attribute name should have been validated before execution.
- validate attributes for WFCORE-3995

**Comment by soul2zimate** [comment]
Hi @bstansberry can this be merged ?

**Comment by soul2zimate** [comment]
retest this please

File: domain-management/src/main/java/org/jboss/as/domain/management/access/ApplicationClassificationConfigResourceDefinition.java [hunk 1]
```diff
@@ -121,17 +119,29 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
 
         @Override
         public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
-            final String attribute = operation.require(NAME).asString();
+            PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
```
Comment (bstansberry) [inline domain-management/src/main/java/org/jboss/as/domain/management/access/ApplicationClassificationConfigResourceDefinition.java]: If you need the address of the current op, use 'context.getCurrentAddress()'.

I don't think you need it though; see my next comment.

File: domain-management/src/main/java/org/jboss/as/domain/management/access/ApplicationClassificationConfigResourceDefinition.java [hunk 2]
```diff
@@ -121,17 +119,29 @@ public void execute(OperationContext context, ModelNode operation) throws Operat
...
+            ModelNode modelNode = context.readResourceFromRoot(address).getModel();
+            // record model values for rollback handler
+            ModelNode configuredApplication = modelNode.get(ModelDescriptionConstants.CONFIGURED_APPLICATION);
```
Comment (bstansberry) [inline domain-management/src/main/java/org/jboss/as/domain/management/access/ApplicationClassificationConfigResourceDefinition.java]: These lines can come after L128 and the first line becomes

ModelNode modelNode = resource.getModel();

The Resource from context.readResourceForUpdate includes the data that you need to restore.

**Comment by soul2zimate** [comment]
Thanks, I have updated this one.

**Pull request merged**